### PR TITLE
ts-migration/consistent-test-it

### PR DESCRIPTION
**Pull request by G-Rath**

### CODE REVIEW

File: src/rules/consistent-test-it.ts [hunk 1]
```diff
@@ -26,7 +30,14 @@ export default {
         additionalProperties: false,
       },
     ],
+    type: 'suggestion',
   },
+  defaultOptions: [
+    { fn: 'test', withinDescribe: 'it' } as {
+      fn?: string;
```
Comment (SimenB) [inline src/rules/consistent-test-it.ts]: can we use the enum instead of `string`?

File: src/rules/consistent-test-it.ts [hunk 2]
```diff
@@ -39,6 +50,10 @@ export default {
       CallExpression(node) {
         const nodeName = getNodeName(node.callee);
 
+        if (!nodeName) {
+          return;
```
Comment (SimenB) [inline src/rules/consistent-test-it.ts]: new, but seeing as tests pass I guess it doesn't matter.

Comment (G-Rath) [reply src/rules/consistent-test-it.ts]: It's required by typescript - `getNodeName` could always return null:

```
export const getNodeName = node => {
  function joinNames(a, b) {
    return a && b ? `${a}.${b}` : null;
  }

  switch (node && node.type) {
    case 'Identifier':
      return node.name;
    case 'MemberExpression':
      return joinNames(getNodeName(node.object), getNodeName(node.property));
  }

  return null;
};
```

Ah, the joys of complex ASTs :joy:

Comment (G-Rath) [reply src/rules/consistent-test-it.ts]: > seeing as tests pass I guess it doesn't matter.

I believe this is b/c of the `isDescribe` - the conditions for which `getNodeName` would return `null` are mutually exclusive w/ `isDescribe`, which is why this has never been a problem.

File: src/rules/consistent-test-it.ts [hunk 3]
```diff
@@ -26,7 +30,14 @@ export default {
         additionalProperties: false,
       },
     ],
+    type: 'suggestion',
   },
+  defaultOptions: [
+    { fn: 'test', withinDescribe: 'it' } as {
+      fn?: 'it' | 'test';
```
Comment (SimenB) [inline src/rules/consistent-test-it.ts]: do we need `as const` on the schema?

Comment (SimenB) [reply src/rules/consistent-test-it.ts]: might not need the type info here then

Comment (G-Rath) [reply src/rules/consistent-test-it.ts]: Not as far as I'm aware - the `schema` is json schema, so it's not actually used by TS (it'd be cool if it was, but it's far too complex)

File: src/rules/consistent-test-it.ts (hunk 2 above)
Comment (G-Rath) [reply src/rules/consistent-test-it.ts]: Just so it's written down somewhere, this is the code that can be used to make `getNodeName` to return `null`:

```
[1,2,3].forEach(() => { test("foo") })
```

It's b/c `[1,2,3]` is an `ArrayExpression`, which causes `getNodeName` to bail out.

I needed to know how to do this to implement a test for another rule to get coverage other the line :)

**Pull request merged**